lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -365,10 +365,12 @@ class DeadFunctionElimination : FunctionLivenessComputation {
   /// Removes all dead methods from vtables and witness tables.
   void removeDeadEntriesFromTables() {
     for (SILVTable &vTable : Module->getVTableList()) {
-      vTable.removeEntries_if([this](SILVTable::Pair &entry) -> bool {
+      vTable.removeEntries_if([&](SILVTable::Pair &entry) -> bool {
         if (!isAlive(entry.second)) {
           DEBUG(llvm::dbgs() << "  erase dead vtable method " <<
                 entry.second->getName() << "\n");
+          // Update cache:
+          Module->removeFromVTableCache(&vTable, entry.first);
```

Comment:
And if SILVTable::removeEntries_if could have cleanly used the Module I wouldn't have written my previous solution in the first place nor added this public function now: SILVTable is friend class of SILModule :)